builtin/providers/aws/resource_aws_codecommit_repository.go

```diff
@@ -104,7 +104,7 @@ func resourceAwsCodeCommitRepositoryCreate(d *schema.ResourceData, meta interfac
 func resourceAwsCodeCommitRepositoryUpdate(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).codecommitconn
 
-	if d.HasChange("default_branch") {
+	if d.HasChange("default_branch") && len(d.Get("default_branch").(string)) > 0 {
```

Comment:
Not sure what you mean, because length bigger than 0 means the same than checking non-empty string.

Anyway, checking other code in terraform I found other examples where they use the code `d.GetOK()`, so I refactored it to use that.

I also added a commit to check the remote repository before updating the default branch. If the default branch remotely is nil, that means that the repository has no branches and any branch update will fail. 

Effectively this means that `default_branch` parameter will only apply if there are any branches in the remote repository. This solves the issue of creating a new repository with `default_branch` defined.